src/__test__/unit/index.test.ts

```diff
@@ -88,11 +89,87 @@ describe('Condition Engine', () => {
       { expression: [OPERATOR_NOR]},
       { expression: [OPERATOR_XOR]},
     ]
-  
+
     for (const exception of exceptions) {
       // @ts-ignore
       expect(() => engine.parse(exception.expression))
         .toThrowError()
     }
   })
+
+  describe('successful evaluate', () => {
+    const testCases = [{
+      name: 'OVERLAP',
+      condition: [
+        'OVERLAP',
+        [
+          '$State1',
+          '$State2',
+          '$State3',
+          '$State4'
+        ],
+        [
+          'TX',
+          'CA'
+        ]
+      ],
+      inputs: [{
+        data: {
+          State1: 'TX'
+        },
+        expected: true
+      }, {
+        data: {
+          State1: 'AL'
+        },
+        expected: false
+      }, {
+        data: {
+          State1: 'TX',
+          State2: 'AL',
+          State3: 'CA',
+          State4: 'CO'
+        },
+        expected: true
+      }, {
+        data: {
+          State1: 'MI',
+          State2: 'RI',
+          State3: 'NY',
+          State4: 'NY'
+        },
+        expected: false
+      }]
+    }, {
+      name: 'UNDEFINED',
+      condition: [
+        'UNDEFINED',
+        '$Name'
+      ],
+      inputs: [{
+        data: {
+          Name: undefined
+        },
+        expected: true
+      }, {
+        data: {
+          Name: 'David'
```

Comment:
😄 ... it looks like I've set it up for myself

Reply:
🤣